### PR TITLE
Fix/compatibility with firefox

### PR DESCRIPTION
**Pull request by HenriChabert**

### CODE REVIEW

**Comment by andreybavt** [comment]
@HenriChabert , could you please remind me what was incompatibility related to?

**Comment by HenriChabert** [comment]
@andreybavt The incompatibility comes from the fact that Chrome and Firefox/Safari handle the doubleclick differently :
- When dblclick, Chrome takes only the div you are clicking on, so it is pretty easy to get all the div and surrounding it by something
- For Firefox, it takes the div and anchor goes at the end of the previous element, which is problematic because the previous element is not always the same, and because what my code does is selecting entirely div that are partially selected, so if the anchor goes at the end of previous word, the previous word will be selected

If you want to understand better what I say, try creating the following html page : 
```html
<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="utf-8">
    <title></title>
  </head>
  <body>
    <span>this </span><span>is </span><span>it</span>
  </body>
</html>
```

In both in Chrome and firefox, double click on the word "is", go to the developer console and use the following instruction : `document.getSelection().getRangeAt(0)`.
You will see for Chrome:
image

And for Firefox:
image

They do not share the same startContainer, and this is very annoying!

The real fix is at line 168, what I basically did is checking whether my startContainer offset is the same as the startContainer length. If so, it means my startContainer is at the previous word and I should take the next one.

**Review by andreybavt — COMMENTED** [review]
Generally good for me apart from small comments

File: resource/labeling-app/components/text-sample/textArea.js [hunk 1]
```diff
@@ -154,14 +157,25 @@ const TextArea = {
...
             })
         },
-        handleMouseUp() {
+        findRangeBoundaries() {
             const selection = document.getSelection();
             if (selection.isCollapsed || selection.toString() === this.tokenSep) return;
-            const range = selection.getRangeAt(0);
-            let [startNode, endNode] = [range.startContainer, range.endContainer]
+            const range = selection.getRangeAt(selection.rangeCount - 1);
+            let [startNode, endNode] = [
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply resource/labeling-app/components/text-sample/textArea.js]: Fixed in a274e65

File: resource/labeling-app/components/text-sample/textArea.js [hunk 2]
```diff
@@ -154,14 +157,25 @@ const TextArea = {
...
+                range.startContainer.nodeType === 1 ? range.startContainer.childNodes[0] : range.startContainer,
```
[the remark this answers is not in the document]
Comment (HenriChabert) [reply resource/labeling-app/components/text-sample/textArea.js]: Fixed in 157f25e

**Review by andreybavt — APPROVED** [review]

**Comment by HenriChabert** [comment]
Closed because I created a new one adapted to spacy tokenization

**Pull request closed without merging**